runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -997,6 +1003,90 @@ private static void translateFn(
     }
   }
 
+  /**
+   * Builds a {@link RunnerApi.PTransform} with adequate context to supply all the information
+   * needed for the Dataflow worker to execute the {@link DoFn} without having to inspect a larger
+   * subgraph.
+   */
+  private static byte[] payloadForFn(
```

Comment:
I would much prefer this return the `MessageWithComponents` than an opaque `byte[]`; The translating to bytes is pretty straightforwards